### PR TITLE
Beta to live

### PR DESCRIPTION
**Pull request by tomdoughty**
## Description

Use nhsuk-frontend library from NPM package
Update views to use Nunjucks templates from nhsuk-frontend
Use relevant SCSS for components from nhsuk-frontend
Use compiled JS from nhsuk-frontend
Remove redundant JS and SCSS which is replaced by nhsuk-frontend
Update UI to follow patterns documented in the service manual
Update site root to include /service-search for consistency with live services

All planned work needed for moving service from Beta to Live. To be reviewed in CI environment by stakeholders. And tested by developers before review and move to higher environments. 

## Motivation and Context

The application needed the UI to be updated in preparation for moving onto the nhs.uk domain
As part of this it was decided to put all live finder services onto the /service-search path

## Checklist

- [x] An [ADR](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions)
- [x] New and/or updated tests
- [x] Changes log in `CHANGELOG`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+0.23.0 / TBC
```
Comment (st3v3nhunt) [inline CHANGELOG.md]: Seems like the date of release is know. If that is the case, can this be updated accordingly?

File: test/integration/checkPage.js [hunk 2]
```diff
@@ -21,16 +21,15 @@ describe('Check page', () => {
   });
 
   it('has a link to the next page', () => {
-    expect($('.button__check').text()).to.equal('Next');
-    expect($('.button__check').prop('href')).to.equal(`${constants.siteRoot}${routes.search.path}`);
+    expect($('.nhsuk-button').text()).to.contain('Continue');
```
Comment (st3v3nhunt) [inline test/integration/checkPage.js]: Why has `equal` been replaced with `contain` - does it need to change?

File: test/integration/checkPage.js [hunk 3]
```diff
@@ -21,16 +21,15 @@ describe('Check page', () => {
...
+    expect($('.nhsuk-button').prop('href')).to.contain(`${constants.siteRoot}${routes.search.path}`);
```
Comment (st3v3nhunt) [inline test/integration/checkPage.js]: Why has `equal` been replaced with `contain` - does it need to change?

File: test/integration/searchPage.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ describe('Search page', () => {
     });
 
     it('has a link to the next page', () => {
-      expect($('.button__search').val()).to.equal('Find');
+      expect($('.nhsuk-button').text()).to.contain('Continue');
```
Comment (st3v3nhunt) [inline test/integration/searchPage.js]: Why has `equal` been replaced with `contain` - does it need to change?

File: test/integration/startPage.js [hunk 5]
```diff
@@ -14,22 +15,21 @@ describe('Start page', () => {
...
-    const res = await chai.request(server).get(`${constants.siteRoot}`);
+    const res = await chai.request(server).get(`${constants.siteRoot}${routes.start.path}`);
 
     $ = cheerio.load(res.text);
   });
 
   it('has a link to the next page', () => {
-    expect($('.button__start').text()).to.equal('Find help');
-    expect($('.button__start').prop('href')).to.equal(`${constants.siteRoot}${routes.check.path}`);
+    expect($('.nhsuk-button').text()).to.contain('Start');
```
Comment (st3v3nhunt) [inline test/integration/startPage.js]: Why has `equal` been replaced with `contain` - does it need to change?

File: test/lib/expectations.js [hunk 6]
```diff
@@ -10,20 +10,20 @@ function htmlWithStatus(res, status) {
...
+  expect($('.nhsuk-breadcrumb__item').eq(1).text().trim()).to.equal('Services near you');
+  expect($('.nhsuk-breadcrumb__item').eq(1).find('a').prop('href')).to.equal('https://www.nhs.uk/service-search');
 }
 
 // eslint-disable-next-line no-script-url
-function backLinkContent($, href = 'javascript:history.go(-1)', text = 'Back', selector = '.link-back') {
-  expect($(selector).prop('href')).to.equal(href);
-  expect($(selector).text()).to.equal(text);
+function backLinkContent($, href = 'javascript:history.go(-1)', text = 'Go back', selector = '.nhsuk-back-link__link') {
+  expect($(selector).prop('href')).to.contain(href);
```
Comment (st3v3nhunt) [inline test/lib/expectations.js]: Why has `equal` been replaced with `contain` - does it need to change?

Comment (tomdoughty) [reply test/lib/expectations.js]: @st3v3nhunt Using the macros from the library annoyingly fills the body with `\n`. I am going to fix this in the frontend library as I believe it is an issue in their code not in ours. Alternatively I could try to clean the body used by our tests

Comment (st3v3nhunt) [reply test/lib/expectations.js]: Ok, good to know 👍 
I won't be a pedant and suggest the value is `trimmed`!

**Comment by st3v3nhunt** [comment]
LGTM. Happy for you to merge when you are ready @tomdoughty

**Pull request merged**